### ISSUE TITLE
torch.inner raises segmentation fault when the operation is on an empty + float16 tensor

### ISSUE THREAD

**Issue by drewshark** [pytorch/pytorch#121148]
### 🐛 Describe the bug

Not sure if it is a duplicate of previous ones: https://github.com/pytorch/pytorch/issues/115068, https://github.com/pytorch/pytorch/issues/115066.

When torch.inner is operated on an empty+float16 tensor, the program will directly raise a segmentation fault

```
import torch
import numpy as np
input = torch.tensor(27)
other = torch.tensor(np.random.rand(6,0), dtype=torch.half)
torch.inner(torch.tensor(27), torch.tensor(np.random.rand(6,0), dtype=torch.half))
```

### Versions

Versions of relevant libraries:
[pip3] numpy==1.25.2
[pip3] torch==2.1.0+cu121
[pip3] torchaudio==2.1.0+cu121
[pip3] torchdata==0.7.0
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.16.0
[pip3] torchvision==0.16.0+cu121
[pip3] triton==2.1.0

**Comment by malfet** [comment]
Indeed a duplicate that is already fixed in 2.1.2(see https://github.com/pytorch/pytorch/pull/116036 ) as well as in 2.2.x release.
Is there a reason why you must stick to 2.1.0?

**Comment by drewshark** [comment]
No, thanks for the information that it has been fixed. I will try 2.2.x later.
